Automattic/wp-calypso
# Paid Stats: Gate time interval selector

Fixes https://github.com/Automattic/dotcom-forge/issues/4953

## Proposed Changes

* Added gating to interval dropdown selector
* When clicked, should redirect to PWYW screen


https://github.com/Automattic/wp-calypso/assets/6586048/0bdd90eb-1bc8-49c8-8279-38d58710ef3b



## Testing Instructions

Enabled by default in development, in calypso live enable with the query param: `?flags=stats/paid-wpcom-v2`
Disable paid stats in development with the query param: `?flags=-stats/paid-wpcom-v2`

* View stats in calypso.live and jetpack, ensure the feature locking in the video above isn't triggered.
* Go to /stats
* Purchase the PWYW to see if it unlocks the gating
* Make sure track events are logged
* Check stats in `/store` to make sure to not break anything (they share the `StatsPeriodNavigation` component) 

## Pre-merge Checklist
